### PR TITLE
roachprod: support deploying local binaries

### PR DESCRIPTION
**Pull request by DarrylWong**
Users can now use roachprod deploy to deploy a
locally built cockroach binary.

Release note: none
Epic: none
Fixes: none

### CODE REVIEW

File: pkg/cmd/roachprod/main.go [hunk 1]
```diff
@@ -778,12 +778,14 @@ const (
...
                ./scripts/tag-custom-build.sh. Must provide a specific tag.`
+	localApp = `
+  local      - Use a provided local binary, must provide the path to the binary.`
 	workloadApp = `
   workload   - Cockroach workload application.`
 )
 
 var deployCmd = &cobra.Command{
-	Use:   "deploy <cluster> <application> <version>",
+	Use:   "deploy <cluster> <application> <version>/<binary-path>",
```
Comment (srosenberg) [inline pkg/cmd/roachprod/main.go:788]: Nit: use `|` to denote an alternative; also `binary-path` -> `pathToBinary`

File: pkg/cmd/roachprod/main.go [hunk 2]
```diff
@@ -794,15 +796,21 @@ stage it on the cluster. It will then perform a rolling upgrade of the cluster,
...
-  %s`, strings.TrimSpace(cockroachApp+releaseApp+customizedApp)),
+  %s`, strings.TrimSpace(cockroachApp+releaseApp+customizedApp+localApp)),
 	Args: cobra.RangeArgs(2, 3),
 	Run: wrap(func(cmd *cobra.Command, args []string) error {
 		versionArg := ""
-		if len(args) == 3 {
+		binaryPath := ""
+		if args[1] == "local" {
+			if len(args) < 3 {
+				return errors.New("local application requires a binary path: deploy <cluster> local <binary-path>")
```
Comment (srosenberg) [inline pkg/cmd/roachprod/main.go:806]: Nit: "...requires a path to the binary" and `binary-path` -> `pathToBinary`.

**Review by srosenberg — APPROVED** [review]

File: pkg/cmd/roachprod/main.go (hunk 1 above)
Comment (DarrylWong) [reply pkg/cmd/roachprod/main.go]: Done.

File: pkg/cmd/roachprod/main.go (hunk 2 above)
Comment (DarrylWong) [reply pkg/cmd/roachprod/main.go]: Done.

**Comment by DarrylWong** [comment]
TFTR!

bors r=srosenberg

**Pull request merged**